### PR TITLE
DOC-1498: Rename imagetools and toc plugins

### PR DESCRIPTION
**Pull request by EkimChau**
Related Ticket: 

Description of Changes:
* Placeholder text

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed (New product features only) Release Note added
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/partials/configuration/editimage_api_key.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+== `+editimage_api_key+`
+
+Placeholder
```
Comment (EkimChau) [inline modules/ROOT/partials/configuration/editimage_api_key.adoc:3]: DOC-1574

File: modules/ROOT/partials/configuration/editimage_upload_timeout.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+== `+editimage_upload_timeout+`
+
+Placeholder
```
Comment (EkimChau) [inline modules/ROOT/partials/configuration/editimage_upload_timeout.adoc:3]: DOC-1574

**Review by tiny-james — APPROVED** [review]

File: modules/ROOT/nav.adoc [hunk 3]
```diff
@@ -332,6 +333,7 @@
 ***** xref:tinydrive-pick.adoc[The Pick API]
 ***** xref:tinydrive-type-interfaces.adoc[TypeScript interfaces]
 **** xref:tinydrive-changelog.adoc[Changelog]
+*** xref:tableofcontents.adoc[Table of Contents]
```
Comment (lnewson) [inline modules/ROOT/nav.adoc:336]: This needs moving so that it's in alphabetical order.

- [ ] Move to be before the `Tiny Drive` plugin docs

File: modules/ROOT/pages/editimage.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
+= Edit Image
+
+:title_nav: Edit Image
+:description: Image editing features for TinyMCE.
+:keywords: edit image rotate rotateleft rotateright flip flipv fliph editimage Edit Image options
```
Comment (lnewson) [inline modules/ROOT/pages/editimage.adoc:5]:
```suggestion
:keywords: edit image rotate rotateleft rotateright flip flipv fliph editimage imageoptions
```
`edit` and `image` are already in the keywords. Also `options` doesn't really make sense and the original `imageoptions` makes more sense to me since this relates to images so I've reverted that.

File: modules/ROOT/pages/editor-command-identifiers.adoc [hunk 5]
```diff
@@ -321,12 +321,12 @@ The following command requires the xref:image.adoc[Image (`+image+`)] plugin.
...
-[[imagetools]]
-==== Image Tools
+[[editimage]]
+==== Edit Image
 
-The following commands require the xref:imagetools.adoc[Image Tools (`+imagetools+`)] plugin.
+The following commands require the xref:editimage.adoc[Edit Image (`+editimage+`)] plugin.
 
-include::partial$commands/imagetools-cmds.adoc[leveloffset=+3]
+include::partial$commands/editimage-cmds.adoc[leveloffset=+3]
```
Comment (lnewson) [inline modules/ROOT/pages/editor-command-identifiers.adoc:329]: This and the toc entry need moving to be in alphabetical order

- [ ] Move to the correct location and update the toc at the top of the page.

File: modules/ROOT/partials/configuration/editimage_api_key.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+== `+editimage_api_key+`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/editimage_api_key.adoc:1]: This is meant to be an internal/cloud only option from my understanding, so I don't believe this should be documented. If it's not reverted then this should have an id set like so: `[[editimage_api_key]]`

- [ ] This file addition should probably be reverted

File: modules/ROOT/partials/configuration/editimage_upload_timeout.adoc [hunk 7]
```diff
@@ -0,0 +1,17 @@
+== `+editimage_upload_timeout+`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/editimage_upload_timeout.adoc:1]:
```suggestion
[[editimage_upload_timeout]]
== `+editimage_upload_timeout+`
```
All options should have a fixed id.

File: modules/ROOT/partials/configuration/icon_list.adoc [hunk 8]
```diff
@@ -163,7 +163,7 @@
 |`+template+` |image:icons/template.svg[template.svg] |`+template.svg+`
 |`+temporary-placeholder+` |image:icons/temporary-placeholder.svg[temporary-placeholder.svg] |`+temporary-placeholder.svg+`
 |`+text-color+` |image:icons/text-color.svg[text-color.svg] |`+text-color.svg+`
-|`+toc+` |image:icons/toc.svg[toc.svg] |`+toc.svg+`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/icon_list.adoc:166]: This wasn't renamed, so this should be reverted: https://github.com/tinymce/tinymce/blob/5c80ea7a2371dacefb9153dd49acb1fdbb9fd1b6/modules/oxide-icons-default/src/svg/toc.svg

- [ ] Revert this change

File: modules/ROOT/partials/module-loading/bundling-plugin-files.adoc [hunk 9]
```diff
@@ -290,9 +290,9 @@ include::partial$plugin-files/plugin-file-list-template.adoc[]
 include::partial$plugin-files/plugin-file-list-textpattern.adoc[]
 
 [[tableofcontentstoc]]
```
Comment (lnewson) [inline modules/ROOT/partials/module-loading/bundling-plugin-files.adoc:292]: This ID should maybe be updated to match the new title?

File: modules/ROOT/partials/toolbar-button-ids/toc-toolbar-buttons.adoc [hunk 10]
```diff
@@ -1,6 +1,6 @@
 [cols="1,3",options="header"]
```
Comment (lnewson) [inline modules/ROOT/partials/toolbar-button-ids/toc-toolbar-buttons.adoc:1]: This file needs renaming to `tableofcontents-toolbar-buttons.adoc`.

- [ ] Rename file to match new plugin name/code

File: modules/ROOT/partials/toolbar-button-ids/toc-toolbar-buttons.adoc [hunk 11]
```diff
@@ -1,6 +1,6 @@
...
 |===
 |Toolbar button identifier |Description
-|`+toc+` |Inserts a Table of Contents into the editor.
+|`+tableofcontents+` |Inserts a Table of Contents into the editor.
 |`+tocupdate+` |Updates the Table of Contents block element.
```
Comment (lnewson) [inline modules/ROOT/partials/toolbar-button-ids/toc-toolbar-buttons.adoc:5]: This needs updating as well looking at the code.
```suggestion
|`+tableofcontentsupdate+` |Updates the Table of Contents block element.
```

File: modules/ROOT/partials/configuration/editimage_api_key.adoc (hunk 6 above)
Comment (EkimChau) [reply modules/ROOT/partials/configuration/editimage_api_key.adoc:2]: those new apis can be reviewed with DOC-1574, i've added the anchors for now

File: modules/ROOT/pages/available-toolbar-buttons.adoc [hunk 12]
```diff
@@ -93,6 +93,12 @@ include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
 :altplugincode: nil
 include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
 
+:plugincategory: opensource
```
Comment (ltrouton) [inline modules/ROOT/pages/available-toolbar-buttons.adoc:96]: I believe this should be this instead
```suggestion
:plugincategory: premium
```

File: modules/ROOT/pages/available-toolbar-buttons.adoc [hunk 13]
```diff
@@ -245,7 +245,7 @@ include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]
 
 :plugincategory: opensource
```
Comment (ltrouton) [inline modules/ROOT/pages/available-toolbar-buttons.adoc:246]:
```suggestion
:plugincategory: premium
```

File: modules/ROOT/partials/configuration/editimage_api_key.adoc (hunk 6 above)
Comment (lnewson) [reply modules/ROOT/partials/configuration/editimage_api_key.adoc:2]: It's poor practice to merge in partial work especially when we know it shouldn't be there, as it just leaves it open to being forgotten/lost later. I'm not comfortable approving this while this still exists, as I'm 99.99% sure this should not be here.

File: modules/ROOT/pages/available-toolbar-buttons.adoc (hunk 12 above)
Comment (lnewson) [reply modules/ROOT/pages/available-toolbar-buttons.adoc]: Fixed

File: modules/ROOT/pages/available-toolbar-buttons.adoc (hunk 13 above)
Comment (lnewson) [reply modules/ROOT/pages/available-toolbar-buttons.adoc:252]: Fixed

**Review by ltrouton — APPROVED** [review]
